index.js

```diff
@@ -0,0 +1,178 @@
+        this.render()
+    }
+
+    onNavigation() {
+        return this.show().catch( this.Error )
+    }
+
+    showNoAccess() {
+        alert("No privileges, son")
+        return this
+    }
+
+    postRender() { return this }
+
+    render() {
+        if( this.data ) this.model = Object.create( this.Model, { } ).constructor( this.data )
+
+        this.slurp( {
+            insertion: this.insertion || { el: document.body },
+            isView: true,
+            template: Reflect.apply( this.template, this.TemplateContext, [ this.getTemplateOptions() ] )
+        } )
+
+        this.els.container.classList.add( this.name )
+        if( this.templateName ) this.els.container.classList.add( this.templateName )
+        if( this.klass ) this.els.container.classList.add( this.klass )
+
+        this.renderSubviews()
+
+        if( this.size ) { this.size(); this.OptimizedResize.add( this.size.bind(this) ) }
+
+        return this.postRender()
+    }
+
+    removeChildren( el ) {
+        while( el.firstChild ) el.removeChild( el.firstChild )
+        return this
+    }
+
+    renderSubviews() {
+        this.subviewElements.forEach( obj => {
+            const name = obj.name || obj.view
+
+            let opts = { }
+
+            if( this.Views && this.Views[ obj.view ] ) opts = typeof this.Views[ obj.view ] === "object" ? this.Views[ obj.view ] : Reflect.apply( this.Views[ obj.view ], this, [ ] )
+            if( this.Views && this.Views[ name ] ) opts = typeof this.Views[ name ] === "object" ? this.Views[ name ] : Reflect.apply( this.Views[ name ], this, [ ] )
+
+            this.views[ name ] = this.Factory.create( obj.view, { insertion: { el: obj.el, method: 'insertBefore' }, ...opts } )
+
+            if( this.events.views ) {
+                if( this.events.views[ name ] ) this.events.views[ name ].forEach( arr => this.views[ name ].on( arr[0], eventData => Reflect.apply( arr[1], this, [ eventData ] ) ) )
+                else if( this.events.views[ obj.view ] ) this.events.views[ obj.view ].forEach( arr => this.views[ name ].on( arr[0], eventData => Reflect.apply( arr[1], this, [ eventData ] ) ) )
+            }
+
+            if( obj.el.classList.contains('hidden') ) this.views[name].hideSync()
+            obj.el.remove()
+        } )
+
+        this.subviewElements = [ ]
+
+        return this
+    }
+
+    async scootAway() {
+        try {
+            await this.Toast.showMessage( 'error', 'You are not allowed here.')
+            this.emit( 'navigate', `/` )
+        } catch( err ) { this.Error( e ); this.emit( 'navigate', `/` ) }
+
+        return this
+    }
+
+    show() { return this.showEl( this.els.container ) }
+
+    showEl( el ) { el.classList.remove('hidden'); return this.animate( el, 'show' ) }
+
+    showSync() { this.els.container.classList.remove('hidden'); return this }
+
+    slurp( opts ) { return Reflect.apply( this.Slurper.slurpTemplate, this, [ this.els, this.events, this.subviewElements, opts ] ) }
```

Comment:
I do not like this pattern.  If slurping things are to be sectioned into their own thing, which is fine, it should be more agnostic.  If its going to be one function you just call, then:

`const { els, events, subviewElements } = this.Sluper.slurpTemplate( template, options );`

Having to call it with `this` binds `View` and `Slurper` as they always have been, so just putting it into its own file, but having to use `this` referencing things in another file is not something I want to continue doing.